src/core/request_handler.rs

```diff
@@ -1,27 +1,19 @@
 // Copyright 2020 Contributors to the Parsec project.
 // SPDX-License-Identifier: Apache-2.0
-#![allow(dead_code)]
-
+use super::IpcHandler;
 use crate::error::{ClientErrorKind, Result};
 #[cfg(test)]
 use mockstream::SyncMockStream;
 use parsec_interface::requests::{Request, Response};
-use std::io::{Read, Write};
-#[cfg(not(test))]
-use std::os::unix::net::UnixStream;
-use std::path::PathBuf;
-use std::time::Duration;
 
 const DEFAULT_MAX_BODY_SIZE: usize = usize::max_value();
-const DEFAULT_SOCKET_PATH: &str = "/tmp/security-daemon-socket";
 
 /// Low level client structure to send a `Request` and get a `Response`.
 #[derive(Clone)]
 #[cfg_attr(not(test), derive(Debug))]
 pub struct RequestHandler {
     pub max_body_size: usize,
-    pub timeout: Option<Duration>,
-    pub socket_path: PathBuf,
+    pub ipc_handler: IpcHandler,
     #[cfg(test)]
     pub mock_stream: SyncMockStream,
```

Comment:
btw that would not be needed with the abstraction that you have currently, right?